l2geth/core/state_transition_ovm.go

```diff
@@ -100,6 +100,11 @@ func EncodeSimulatedMessage(msg Message, timestamp, blockNumber *big.Int, execut
 		to = &common.Address{0}
 	}
 
+	value := msg.Value()
+	if value == nil {
+		value = big.NewInt(0)
+	}
+
```

Comment:
Ah I see this is only for simulated messages (eth_call) not for normal txs  -- that right?

Reply:
yes, see line 97 (`func EncodeSimulatedMessage...`)